### ISSUE TITLE
Beilstein feedback on SVN revision 2210-2213

### ISSUE THREAD

**Issue by jbs1** [jbs1/tract2#183]
_migrated from Trac, where originally posted by **beilstein** on 6-Sep-2013 2:18pm_

Many of the manuscripts in the repository were converted without errors. Thanks for the great work. However, some errors occured (very few, but at repeated instances), and they are reported here.

A few general remarks at the beginning. For the sake of consistency, all tex sources were consistently renamed to the scheme mXXXXXX.tex for processing. Some of these sources needed to be modified and were subsequently re-comitted to the repository. Hence, the SVN revision is meanwhile 2213. If you need to reproduce the errors, please use the mXXXXXXX.tex sources in each directory.

Instead of going through each manuscript separately, I decided to sort the errors by type and to give the respective manuscript numbers. If possible, I also try to point out, where the problem might be found.

1. Two manuscripts failed conversion to .beil. Unfortunately, I could not find out where the problem was from reading the log files. However, a copy of the log files are in the respective directory with a bracketed (Beilstein) in the file name. 

m2926664 m3592508

2. The attribute value "" of type ID in the element <reference> is empty. This happens, when there is a line break in a string in the \bib-item-argument. A "%" is inserted, which presumably causes the problem. For instance in *m1081239*:
```
\setboolean{nobreakdashused}{false}\bibitem[Ehrhart(1991)]{landolt_three_twent%
yfive}
Ehrhart,~P. Atomic Defects in Metals. In \emph{Landolt-B\"ornstein: Numerical
  data and functional relationships in science and technology}; Ullmaier,~H.,
  Madelung,~O., Hellwege,~K.-H., Eds.;
\newblock Group III, Condensed Matter, Vol. 25;
\newblock Springer-Verlag: Heidelberg, Germany, 1991.
```
is converted to
```
 <reference type="unknown" id="">
      <comment>
         <paragraph>
            <chunk> Ehrhart, P. Atomic Defects in Metals. In </chunk>
            <chunk italic="yes">Landolt-Börnstein: Numerical data and functional relationships in science and technology</chunk>
            <chunk>; Ullmaier, H., Madelung, O., Hellwege, K.-H., Eds.; </chunk>
            <chunk>Group III, Condensed Matter, Vol. 25; </chunk>
            <chunk>Springer-Verlag: Heidelberg, Germany, 1991. </chunk>
         </paragraph>
      </comment>
   </reference>
```

m1081239 m1472120 m1475912 m1541500 m1698334 m1715028 m1872174  m1952193 m2027014 m2081680 m2525915 m2960319 m3184105 m3332599 m3539697 m3566801 m3577250

2. The LTXML tags <p role="background">,  <p role="results"> and <p role="conclusion"> are not converted into XML appropriately. Hence, the bold faced expressions "Background:", "Results:" and "Conclusion:" are missing in the abstract section. See 

https://trac.kwarc.info/arXMLiv/ticket/1529#comment:6

m1124044 m1140600 m1242256 m1473569 m1698392 m2336656 m2390089 m2445518 m2591962 m3121575 m3479303 

3. In *m1473569*, the citation to Equation 3 is not converted. 
```
full polaron transformed compound (Equation \ref{HPMTpol}). Then[…]
```
is converted to
```
<chunk>To derive an equation of motion for the combined dynamics of charge and phonon degrees of freedom, one starts from a Liouville–von Neumann equation for the full polaron transformed compound (Equation </chunk>            <!--Elided element ref --><chunk>). Then […]
```

```
where all expectation values are calculated with the full Hamiltonian (Equation \ref{HPMTpol})
```
is converted to
```
<chunk>where all expectation values are calculated with the full Hamiltonian (Equation </chunk>
            <!--Elided element ref --><chunk>). Of course, for </chunk>
```

```
the model (Equation \ref{HPMTpol}) can 
```
is converted to
```
<chunk>, the model (Equation </chunk>            <!--Elided element ref --><chunk>) can</chunk>
```

4. In *m1664701*, Figure 1 is converted into a <float-target> with an empty attribute "id".[[BR]]
LaTeX:
```
Consider a flux of electrons travelling through an elastic medium. Intuitively, we expect
the flux to preferentially emit phonons with momentum parallel to the electron flow.
This becomes evident, when we consider the setup in figure 1.

\begin{figure}
\begin{center}
\includegraphics{m1664701-1}
\caption{An electrode--junction--electrode system, in the Landauer picture.
The details are discussed in the text.}
\end{center}
\end{figure}
```

XML:
```
<chunk>Consider a flux of electrons travelling through an elastic medium. Intuitively, we expect the flux to preferentially emit phonons with momentum parallel to the electron flow. This becomes evident, when we consider the setup in figure 1.</chunk>
         </paragraph>
         <float target=""/>
```

5. Author affiliations are not transformed correctly. For instance, in *m1670316*:

LaTeX:
```
\author{F. May}
\affiliation{Institut f\"ur Theoretische Physik, Johann Wolfgang Goethe-Universit\"at, 60438 Frankfurt/Main, Germany}
\affiliation{Max Planck Institute for Polymer Research, 55128 Mainz, Germany}

\author*{M. R. Wegewijs}{m.r.wegewijs@fz-juelich.de}
\affiliation{Institut f\"ur Theorie der Statistischen Physik, RWTH Aachen, 52056 Aachen,  Germany}
\affiliation{Peter Gr{\"u}nberg Institut and JARA - Fundamentals of Information Technology, Forschungszentrum J{\"u}lich, 52425 J{\"u}lich,  Germany }
\author[1,2]{W. Hofstetter}
```

LTXML:
```
<creator role="author">
    <contact role="name">F. May</contact>
    <contact mark="1" role="affiliation">Institut für Theoretische Physik, Johann Wolfgang Goethe-Universität, 60438 Frankfurt/Main, Germany</contact>
    <contact mark="2" role="affiliation">Max Planck Institute for Polymer Research, 55128 Mainz, Germany</contact>
  </creator>
  <creator role="corresponding-author">
    <contact role="name">M. R. Wegewijs</contact>
    <contact role="address">m.r.wegewijs@fz-juelich.de</contact>
    <contact mark="3" role="affiliation">Institut für Theorie der Statistischen Physik, RWTH Aachen, 52056 Aachen,  Germany</contact>
    <contact mark="4" role="affiliation">Peter Grünberg Institut and JARA - Fundamentals of Information Technology, Forschungszentrum Jülich, 52425 Jülich,  Germany </contact>
  </creator>
  <creator role="author">
    <contact role="name">W. Hofstetter</contact>
    <contact mark="1,2" role="affiliation"/>
  </creator>
```

XML:
```
  <author given-names="" last-name="F. May" email="" affiliations="a1 a2"/>
   <author given-names="" last-name="M. R. Wegewijs" email="m.r.wegewijs@fz-juelich.de"
           corresponding-author="yes"
           affiliations="a3 a4"/>
   <author given-names="" last-name="W. Hofstetter" email="" affiliations="a5"/>
   <affiliation id="a1">Institut für Theoretische Physik, Johann Wolfgang Goethe-Universität, 60438 Frankfurt/Main, Germany</affiliation>
   <affiliation id="a2">Max Planck Institute for Polymer Research, 55128 Mainz, Germany</affiliation>
   <affiliation id="a3">Institut für Theorie der Statistischen Physik, RWTH Aachen, 52056 Aachen,  Germany</affiliation>
   <affiliation id="a4">Peter Grünberg Institut and JARA - Fundamentals of Information Technology, Forschungszentrum Jülich, 52425 Jülich,  Germany </affiliation>
   <affiliation id="a5"/>
```

m1670316 m2018234 m2458960 m2806550 m3121575 m3539471 m3764443

6. In *m1771034*, the table footnotes are not converted in footnotes. (In Beilstein-XML, table footnotes are supposed to be chunks in the element <table-footer>, see page 25 ff. of Creating_Beilstein_XMLFile_20130607.pdf)

7. Also in *m1771034*, the BBL-code
```
\setboolean{nobreakdashused}{false}\bibitem[exp()]{exp_butane_homo}
 \url{http://webbook.nist.gov/cgi/cbook.cgi?ID=C106978&Units=SI&Mask=20#Ion-En%ergetics} (accessed Aug 16, 2011).
```
gives a wrongly formatted reference (again due to a "%"):

XML
```
 <reference type="unknown" id="r55">
      <comment>
         <paragraph>
            <chunk> </chunk>
            <chunk>http://webbook.nist.gov/cgi/cbook.cgi?ID=C106978&amp;Units=SI&amp;Mask=20#Ion-En%Ωergetics</chunk>
            <chunk> (accessed Aug 16, 2011). </chunk>
         </paragraph>
      </comment>
      <external-link type="uri"
                     public-id="http://webbook.nist.gov/cgi/cbook.cgi?ID=C106978&amp;Units=SI&amp;Mask=20#Ion-En%ergetics"/>
   </reference>
```

8. In *m1941364*, the unexpected atrribute 'class' (value=ltx_centering) appears in the element <math>. Hence, validation failed. For instance:
```
<inline-math id="im33" fragid="S2.SS1.m15" tex="-v_{y}[\upmu\rm{m/s}](theor.)">
         <m:math  xmlns:m="http://www.w3.org/1998/Math/MathML"
                 alttext="-v_{y}[\upmu\rm{m/s}](theor.)"
                 class="ltx_centering"
                 display="inline">
            <m:mrow>
               <m:mo>-</m:mo>
               <m:msub>
                  <m:mi>v</m:mi>
                  <m:mi>y</m:mi>
               </m:msub>
               <m:mrow>
                  <m:mo>[</m:mo>
                  <m:mi mathvariant="normal">μ</m:mi>
                  <m:mi mathvariant="normal">m</m:mi>
                  <m:mo>/</m:mo>
                  <m:mi mathvariant="normal">s</m:mi>
                  <m:mo>]</m:mo>
               </m:mrow>
               <m:mrow>
                  <m:mo>(</m:mo>
                  <m:mi>theor</m:mi>
                  <m:mo>.</m:mo>
                  <m:mo>)</m:mo>
               </m:mrow>
            </m:mrow>
         </m:math>
      </inline-math>
```

9. In *m2018234*, the references to Supporting Information Files are not converted.

LaTeX:
```
and the scripts are available in \ref{si:Sader} and \ref{si:Matrix}, respectively
```

XML:
```
 <chunk>, and the scripts are available in </chunk>
            <!--Elided element ref --><chunk> and </chunk>
            <!--Elided element ref --><chunk>, respectively
```

The text should read:  "and the scripts are available in Supporting Information File 1 and Supporting Information File 2, respectively"

10. In the converted DOIs, there are strange characters between "doi:" and the number. These characters are either something like a "non-breaking space" or a "capital greek letter omega". Since these characters are not present in the LaTeX-PDF, they should removed from the XML as well.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **beilstein** on 9-Sep-2013 7:55am_

A very serious issue was left out in the last comment.

11. In *m3252448* Schemes are not converted at all. The links are wrong, the targets are missing, in-text references are elided.

LaTeX:
```
\begin{scheme}
\caption{Mechanism explaining the CIDNP effects in sensitized hydrogen abstractions from tertiary aliphatic
amines {\bf DH}.}
\label{scheme:CIDNPMechanism}
\begin{center}
\includegraphics[]{m3252488-i1}
\end{center}
\end{scheme}
```

wrong link in XML:
```
<inline-float target="i1"/>
```
which must be 
```
<float target="s1"/>
```
and no target is given in the XML.

The caption of the Scheme is converted to a normal <paragraph>.

The in-text references are elided:

LaTeX:
```
to show that these reactions are always two-step hydrogen abstractions
according to Scheme \ref{scheme:CIDNPMechanism}.
```

XML:
```
to show that these reactions are always two-step hydrogen abstractions according to Scheme </chunk>
         <!--Elided element ref --><chunk>. </chunk>
```

The same happens with Scheme 2 in the Manuscript.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 13-Sep-2013 12:37pm_

Okay I am going through these errors now. 
First of all, I just updated my repository and I couldn't find new texfiles in all of the folders (specifically in m1081...). 
In that file, I couldn't reproduce the error, the id field has a value.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **beilstein** on 17-Sep-2013 7:15am_

1. Couldn't you find new tex files in _any_ of the folders? Or just in some of them? some of the manuscripts already had the mxxxxxxx.tex naming. For a try I just deleted m1081239.tex and when I updated the repository, the file was restored. So it seems that I committed the files succesfully.

2. Can you reproduce the error in any of the other files?

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 17-Sep-2013 2:55pm_

Replying to [comment:3 beilstein]:
> 1. Couldn't you find new tex files in _any_ of the folders? Or just in some of them? some of the manuscripts already had the mxxxxxxx.tex naming. For a try I just deleted m1081239.tex and when I updated the repository, the file was restored. So it seems that I committed the files succesfully.
> 
I might have just missed it then, it was in the directories when I looked for it. 
> 2. Can you reproduce the error in any of the other files?
 I just looked through all of the .beil files in beilstein_published_manuscripts. There was one single instance if an empty id attribute, in a file I had used for testing. I just updated previously to this. 

For 2.2 (the one with Background etc. ) 
I have fixed the error. Latexml was handling abstract subsections in two different ways. 
for 
```
<p role="background">asdf</p> }}} 
we will get  
{{{
<abstract-subsection>
<title><chunk>Background</chunk></title>
<paragraph>asdf</paragraph>
</abstract-subsection 
``` 
and equivalent things for results and conclusion.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **beilstein** on 17-Sep-2013 3:37pm_

Replying to [comment:4 lkohlhase]:
> Replying to [comment:3 beilstein]:

> > 2. Can you reproduce the error in any of the other files?
>  I just looked through all of the .beil files in beilstein_published_manuscripts. There was one single instance if an empty id attribute, in a file I had used for testing. I just updated previously to this. 

so it might be an issue related to the computer used. when i do the conversion here, a "%" character is inserted into the bbl-file if (and only if) there is a non-blank character at column 80 of a line. for instance in m1472120:
```
\setboolean{nobreakdashused}{false}\bibitem[Coulthard(1967)]{Coulthard_Hartree%
-Fock}
Coulthard,~M.~A. \emph{Proc. Phys. Soc., London} \textbf{1967}, \emph{91},
  44\mynobreakdash 49. \url{doi: 10.1088/0370-1328/91/1/309}.
```
this "%" character obviously causes the empty id value. if you cannot reproduce the error, this would mean, that your linux system is not inserting this character. can you please check your bbl-file of m1472120 to confirm this. thank you. any good idea to get rid of this "%" character would be appreciated very much.

> For 2.2 (the one with Background etc. ) 
> I have fixed the error. Latexml was handling abstract subsections in two different ways. 
> for 
> {{{
> <p role="background">asdf</p> }}} 
> we will get  
> {{{
> <abstract-subsection>
> <title><chunk>Background</chunk></title>
> <paragraph>asdf</paragraph>
> </abstract-subsection 
> }}} 
> and equivalent things for results and conclusion. 

great. i will have a look at this tomorrow.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 17-Sep-2013 4:18pm_

For 3. I think I fixed the issue. 
I had mixed up the ids I was using internally. In my current model I am assuming that math is always referred to by its label.

For 4. I fixed this. Here I wrongly assumed that every figure would have a label ( <==> I was assuming that every figure was being referred to). However this was not the case for this figure. I fixed it by referring to the xml:id attribute every time that we are not in a reference.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 17-Sep-2013 4:21pm_

Replying to [comment:5 beilstein]:
> Replying to [comment:4 lkohlhase]:
> > Replying to [comment:3 beilstein]:
> 
> > > 2. Can you reproduce the error in any of the other files?
> >  I just looked through all of the .beil files in beilstein_published_manuscripts. There was one single instance if an empty id attribute, in a file I had used for testing. I just updated previously to this. 
> 
> so it might be an issue related to the computer used. when i do the conversion here, a "%" character is inserted into the bbl-file if (and only if) there is a non-blank character at column 80 of a line. for instance in m1472120:
> {{{
> \setboolean{nobreakdashused}{false}\bibitem[Coulthard(1967)]{Coulthard_Hartree%
> -Fock}
> Coulthard,~M.~A. \emph{Proc. Phys. Soc., London} \textbf{1967}, \emph{91},
>   44\mynobreakdash 49. \url{doi: 10.1088/0370-1328/91/1/309}.
> }}}
> this "%" character obviously causes the empty id value. if you cannot reproduce the error, this would mean, that your linux system is not inserting this character. can you please check your bbl-file of m1472120 to confirm this. thank you. any good idea to get rid of this "%" character would be appreciated very much.
> 
I just checked the file, no % to be found. I don't really know how to fix that, @Michael do you know?

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 17-Sep-2013 4:23pm_

I fixed the issue for 5.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 17-Sep-2013 4:45pm_

I am not sure how to deal with 6. 
According to the documentation I should essentially insert a text super script into the cell, and then write the table footers below it. 
I think I could implement this, yet it would be difficult to refer to the table footers with letters, would it be possible to use numbers as well?

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 20-Sep-2013 4:40pm_

I fixed 9. however somehow it is not playing nicely with the scaffolds anymore. I didn't change anything about them, but it can't read the file anymore. 
It still finds the file, but if I leave it unchanged, it gets no data from it anymore. But if I remove almost all of teh information and turn it into 
```
<article> random stuff </article> 
``` 
It can read it again. I am trying to find out why it is doing this atm.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **beilstein** on 23-Sep-2013 4:02pm_

i did have a look at the changes you made to fix the issues. i updated to repository version 2217, and i can confirm that many of the errors are fixed. thank you very much.

points 2.1 and 7 (the strange line breaks with the "%") obviously stem from an older bibtex-version. i added a little workaround to the build script, and everything seems fine now.

point 2.2 (background, results, conclusion) now works.

point 3 is fixed.

point 4 is fixed.

point 5 (wrongly converted affiliations) is fine now.

point 8 (illegal value=ltx_centering) seems to stem from an unusual table formatting the authors chose. i will have a look at it again.

still open are the following points:

point 1: the manuscripts still fail conversion

point 6: you understood this correctly. there is superscripted letter in the table cell and the repective text, again with the superscripted letter, in the <table-footer>. this should be implemented with letters.

point 9: you mentioned that you were working on this.

and point 10 and point 11.

can you please give me an estimate about how long you will have to work on these issues? perhaps until the end of this week or until end of next week. i do not want to pressure you, i just need a reliable date to plan with. thank you very much again.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **beilstein** on 23-Sep-2013 4:06pm_

oh, i forgot, there is a typo in the current version of the XSL-file. in the <abstract-subsection> you wrote the element <Title> with uppercase "T", which will generate an error in our HTML renderer. i corrected this to lowercase "t"s in my local working directory. since you are currently editing the XSL-file, you should correct it before committing again. i don't want to interfere with your development. thank you.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 23-Sep-2013 6:24pm_

Replying to [comment:12 beilstein]:
> oh, i forgot, there is a typo in the current version of the XSL-file. in the <abstract-subsection> you wrote the element <Title> with uppercase "T", which will generate an error in our HTML renderer. i corrected this to lowercase "t"s in my local working directory. since you are currently editing the XSL-file, you should correct it before committing again. i don't want to interfere with your development. thank you.

I will probably be able to work 10 and 11 on Friday again. I ought to be able to fix 6, 10 and 11 then. I haven't looked at 1 much yet, so I can't give an estimate there quite yet. 

I will fix the Title thing, thanks for bringing it to my attention. 

The public-ids suddenly not working was also fixed today.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 27-Sep-2013 4:32pm_

I fixed issue #6. 
The table-footers will look like this: 
```
<table-footer>
<paragraph>
<chunk supscript="yes">a</chunk> content </paragraph>
<paragraph><chunk supsript="yes">b></chunk> content for 2 </paragraph>
...
</table-footer>
```

I think I also fixed the title issue.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 27-Sep-2013 4:48pm_

I fixed issue number 10. However, if in any of the references, either of those signs pop up in exactly the fifth position, they will be deleted.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 27-Sep-2013 5:48pm_

I fixed issue number 11. 
The reference that isn't working is not related to this. That reference is referring to the section that the scheme is in. However the current .beil model does not support links to non-graphics/math etc. ( I remember us discussing this). Hence I can't really fix that reference.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 27-Sep-2013 6:02pm_

I just looked at the document from 1. The issue here is not the conversion from .ltxml to .beil, but the conversion from .tex to .ltxml. So I guess Kim should take a look at that. 
Additionally in m3252448, there were empty <cites>, which I guess should be looked at as well.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **beilstein** on 1-Oct-2013 3:12pm_

Replying to [comment:14 lkohlhase]:
> I fixed issue #6. 
> The table-footers will look like this: 
> ```
> <table-footer>
> <paragraph>
> <chunk supscript="yes">a</chunk> content </paragraph>
> <paragraph><chunk supsript="yes">b></chunk> content for 2 </paragraph>
> ...
> </table-footer>
> }}}
> 
> I think I also fixed the title issue. 

unfortunately, there is a <chunk> missing. it must be
{{{
<paragraph><chunk supsript="yes">b></chunk><chunk> content for 2 </chunk></paragraph>
```
i tried to fix it myself in the xsl, because it seemed easy enough to just add <chunk>. however, in this particular case, there are also a citation to a literature reference in the footnotes. but things such as <link target="r12"/> are not allowed in <chunk>. hence, i need you to look at it again.

and please change "supscript" to "superscript". again i could do this myself, but i do not want to create version conflicts in the repository.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **beilstein** on 1-Oct-2013 3:50pm_

Replying to [comment:16 lkohlhase]:
> I fixed issue number 11. 
> The reference that isn't working is not related to this. That reference is referring to the section that the scheme is in. However the current .beil model does not support links to non-graphics/math etc. ( I remember us discussing this). Hence I can't really fix that reference. 

i think there is a misunderstanding here. the reference is definitely pointing to a scheme. the label of the scheme is {scheme:CIDNPMechanism} as can be seem from the tex source above. however, it seems that the conversion from LaTeX to ltxml is not correct. 

there is a
```
 <section refnum="1" xml:id="S1" labels="LABEL:scheme:CIDNPMechanism" fragid="S1">
```
which is simply not correct, because there is no such label to the section in the tex-source. the conversion of the scheme from LaTeX provided this:
```
<para role="scheme" xml:id="S1.p4" fragid="S1.p4">
      <!-- %**** m3252488.tex Line 125 **** -->
      <p><text role="caption">Mechanism explaining the CIDNP effects in sensitized hydrogen abstractions from tertiary aliphatic
amines <text font="bold">DH</text>.</text></p>
      <graphics candidates="m3252488-i1.png" graphic="m3252488-i1" xml:id="S1.p4.g1" class="ltx_centering" fragid="S1.p4.g1" imagesrc="m3252488-i1.png" imagewidth="1299" imageheight="935"/>
    </para>
```
the label is missing here. this must be figured out again, probably by your coworker. 

interestingly enough, in the xml there are the lines
```
<schemes>
      <scheme id="s1">
         <caption>
            <chunk>Mechanism explaining the CIDNP effects in sensitized hydrogen abstractions from tertiary aliphatic amines </chunk>
            <chunk bold="yes">DH</chunk>
            <chunk>.</chunk>
         </caption>
         <graphic public-id="m3252488-i1"/>
      </scheme>
      <scheme id="s2">
         <graphic public-id="m3252488-i2"/>
         <caption>
            <chunk>Pathways from the free radicals to the product.</chunk>
         </caption>
      </scheme>
   </schemes>
```
which seem pretty correct to me. _ONLY_ it must be <caption><paragraph><chunk>... the <paragraph> is missing.

the two xml-lines
```
<scheme external-id="s1" internal-id="S1.p4" label=""/>
   <scheme external-id="s2" internal-id="S2.p10" label=""/>
```
are plain wrong and must be deleted.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **beilstein** on 1-Oct-2013 4:22pm_

Replying to [comment:17 lkohlhase]:
> I just looked at the document from 1. The issue here is not the conversion from .ltxml to .beil, but the conversion from .tex to .ltxml. So I guess Kim should take a look at that. 
> Additionally in m3252448, there were empty <cites>, which I guess should be looked at as well. 

yes, please tell your coworker to have a look at this. i could not find any empty <cites>. can you please tell me, what you have in mind. thank you.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **kpj** on 3-Oct-2013 9:58am_

Hello,
I have created a minimal example, which reproduces the "Excessive recursion" error:
```
\documentclass[]{beilstein}

\usepackage{natbib}

\begin{document}
\begin{thebibliography}{49}

\bibitem[some authors]{Chiang1977}
  \textbf{some text}
\end{thebibliography}
\end{document}
```
It works if `\textbf{some text`} is removed (removing `\usepackage{natbib`} also fixes this problem, but I guess that's not the solution we are looking for...).

Checking with `\show\textbf` yields that `\textbf` is extended to `\protect \textbf`. Indeed, any occurence of `\protect` fails due to the recursion (seems like the error message gave us a good hint :-D ).

My conclusion would be that there is some mistake in the implementation of `\protect`, anyone willing to confirm that?

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 12-Oct-2013 7:54pm_

Okay I fixed the issues that you highlighted. Since Kim hasn't done anything (or at least hasn't let me know about it) about the section error, I couldn't work on that yet. 

Now on revision 2230

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **kohlhase** on 15-Oct-2013 11:26am_

Replying to [comment:1 beilstein]:
> A very serious issue was left out in the last comment.
> 
> 11. In *m3252448* Schemes are not converted at all. The links are wrong, the targets are missing, in-text references are elided.

The explanation is wrong here, the schemes *are* being converted by LaTeXML, only the labeling mechanism was not appliccable to the container `ltx:para` we used for that, and the section element stole the label.  

I have used the float package for that `scheme`, so Lukas can now work with the correct label. This should (soon) fix the remaining issues.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **lkohlhase** on 18-Oct-2013 10:43pm_

Replying to [comment:23 kohlhase]:
> Replying to [comment:1 beilstein]:
> > A very serious issue was left out in the last comment.
> > 
> > 11. In *m3252448* Schemes are not converted at all. The links are wrong, the targets are missing, in-text references are elided.
> 
> The explanation is wrong here, the schemes *are* being converted by LaTeXML, only the labeling mechanism was not appliccable to the container `ltx:para` we used for that, and the section element stole the label.  
> 
I added support for the new implementation of schemes. While doing that, I fixed the link issue, so I think that I handled all the issues of this ticket, apart from the one with the not converting documents. 

Once that is done I think we can close this ticket. 
> I have used the float package for that `scheme`, so Lukas can now work with the correct label. This should (soon) fix the remaining issues.

**Comment by jbs1** [comment]
_migrated from Trac, where originally posted by **kohlhase** on 20-Oct-2013 9:59am_

using the observation of Christian Lange that m2926664 m3592508 work if we remove the `\usepackage{natbib`} I adapted all occurrences of the LaTeX sources and re-ran the newest version of LaTeXML over everything. And indeed, we generate `ltxml` and `beil` files for both of them. So it seems that all problems reported here are fixed.  

BUT there has been a regression with LaTeXML, it develped problems on the `\keywords` macro. I am waiting for Deyan's help on this. Once that is done, I will re-run again. After this is taken care of, I will close the issue and we can start the Endabnahme. 

BTW, there are a coumple of occurrences where `\ref{foo`} has been corrupted to `ef{foo`}, which I fixed. There may be more of those, which I did not find. I am not sure where this comes from. But there may be follow-up problems from the ones I did not find.
